packages/styled-components/src/constructors/createGlobalStyle.js

```diff
@@ -24,98 +23,57 @@ export default function createGlobalStyle(
   ...interpolations: Array<Interpolation>
 ) {
   const rules = css(strings, ...interpolations);
-  const id = `sc-global-${hashStr(JSON.stringify(rules))}`;
-  const style = new GlobalStyle(rules, id);
+  const styledComponentId = `sc-global-${hashStr(JSON.stringify(rules))}`;
+  const globalStyle = new GlobalStyle(rules, styledComponentId);
 
-  class GlobalStyleComponent extends React.Component<GlobalStyleComponentPropsType, *> {
-    styleSheet: Object;
+  function GlobalStyleComponent(props: GlobalStyleComponentPropsType) {
+    const styleSheet = useStyleSheet();
+    const theme = useContext(ThemeContext);
 
-    static globalStyle = style;
+    if (process.env.NODE_ENV !== 'production' && React.Children.count(props.children)) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        `The global style component ${styledComponentId} was given child JSX. createGlobalStyle does not render children.`
```

Comment:
TBH the right thing to do here is to just delete `createGlobalStyle` (breaking this API again...) and add a `useGlobalStyle` instead.

Reply:
You can't currently use hooks in class components, so i'd rather not.